sycl/test/basic_tests/group.cpp

```diff
@@ -1,5 +1,4 @@
-// RUN: %clang --sycl %s -c -o %T/kernel.spv
-// RUN: %clang -std=c++11 -g %s -o %t.out -lstdc++ -lOpenCL -lsycl
+// RUN: %clang -std=c++11 -fsycl %s -o %t.out -lstdc++ -lOpenCL -lsycl
```

Comment:
AFAIK:
- Clang's driver links standard C++ library if option `-std=c++11` is set.
- Clang's driver links SYCL library if option  `-fsycl` is set.
Please, apply to all tests.

```suggestion
// RUN: %clang -std=c++11 -fsycl %s -o %t.out -lOpenCL
```

Reply:
I suggest doing that in a separate patch and only when both Linux and Windows compilers ready to that.

Right now removing -lstdc++ work on Windows (removing it also eliminates one warning there).
But removing -lstdc++ on Linux does not seem to work for me:

    20:47:48 > "/export/iusers/vklochko/ws/gs_public/build/bin/clang" "-I/export/iusers/vklochko/ws/OpenCL-Headers" "-std=c++11" "-fsycl" "/export/iusers/vklochko/ws/gs_public/llvm/sycl/test/basic_tests/accessor/accessor.cpp" "-o" 
"/export/iusers/vklochko/ws/gs_public/build/tools/sycl/test/basic_tests/accessor/Output/accessor.cpp.tmp.out" "-lstdc++" "-lOpenCL" "-lsycl"

    20:48:00 > "/export/iusers/vklochko/ws/gs_public/build/bin/clang" "-I/export/iusers/vklochko/ws/OpenCL-Headers" "-std=c++11" "-fsycl" 
    "/export/iusers/vklochko/ws/gs_public/llvm/sycl/test/basic_tests/accessor/accessor.cpp" "-o" 
    "/export/iusers/vklochko/ws/gs_public/build/tools/sycl/test/basic_tests/accessor/Output/accessor.cpp.tmp.out" "-lOpenCL" "-lsycl"                                                                                                                                                                                            
    /usr/bin/ld: /tmp/accessor-8dc945.o: undefined reference to symbol '_ZNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEaSEPKc@@GLIBCXX_3.4.21'
    //usr/lib/x86_64-linux-gnu/libstdc++.so.6: error adding symbols: DSO missing from command line
    clang-9: error: linker command failed with exit code 1 (use -v to see invocation)

Also, removing -lsycl does not work on Windows (it leads to linkage error) right now.

BTW, You don't suggest removing -lOpenCL when -fsycl is used? Isn't it added automatically with -fsycl?

Reply:
> But removing -lstdc++ on Linux does not seem working for me:

It because some tests use C compiler driver (`clang`) instead of C++ compiler driver (`clang++`).
I suggest fixing this also.

> Also, removing -lsycl does not work on Windows (it leads to linkage error) right now.

This seems like a driver issue.

> BTW, You don't suggest removing -lOpenCL when -fsycl is used? Isn't it added automatically with -fsycl?

No.
I'm not sure it must be done automatically or enabled by some compiler option (e.g. `-fsycl-backend=OpenCL` or `-fsycl-plugin=OpenCL`).

Reply:
I agree with your comments about clang++, removing -lstdc++ and -lsycl and think that those changes deserve a separate big patch. 

Right now ALL tests use 'clang', not 'clang++'. Also, 'clang++' will need a new substitution in lit.cfg. Currently lit.cfg has it for 'clang':
config.substitutions.append( ('%clang', ' ' + config.clang + ' -I'+config.opencl_include ) )

This particular patch fixes few tests with really obsolete compilation commands and enables those tests for Windows.

Reply:
Please send a separate PR.